src/git.rs

```diff
@@ -0,0 +1,48 @@
+use crate::VCS;
+use std::path::PathBuf;
+use std::process::Command;
+
+pub(crate) struct Git();
+
+impl Git {
+    pub(crate) fn new() -> Git {
+        Git()
+    }
+}
+
+impl VCS for Git {
+    fn is_repo(&self, path: &str) -> bool {
+        let mut pathbuf = PathBuf::from(path);
+        pathbuf.push(".git");
+        pathbuf.is_dir()
+    }
+
+    fn commit(&self, msg: &str) -> Result<(), String> {
+        let output = Command::new("git")
+            .arg("commit")
+            .arg("-am")
+            .arg(msg)
+            .output()
+            .map_err(|err| format!("couldn't run git commit: {}", err))?;
+
+        if !output.status.success() {
+            let stdout = String::from_utf8(output.stdout).unwrap_or("(stdout unavailable)".into());
+            let stderr = String::from_utf8(output.stderr).unwrap_or("(stderr unavailable)".into());
+            if stdout.trim() == "nothing changed" {
```

Comment:
It probably is a different message that we need to look for here, I'll fix it up.